### PR TITLE
Replaced Buttons for Select Menu on scsearch

### PR DESCRIPTION
**Pull request by Murky-MoMo**
This pr 

takes out the buttons for the command `scsearch` and implements it, and adds support for further select menu implementations 
(so instead of a \<SlashCommand\>.button method, it uses a `selectmenu` method)

few minor parts of the code were modified too,
like some variables were renamed to be more descriptive and
some _repetitive loops_ were removed

### CODE REVIEW

**Comment by Murky-MoMo** [comment]
This also completes #6

File: slash/music/scSearch.js [hunk 1]
```diff
@@ -4,106 +4,86 @@ require('moment-duration-format');
...
-			let array = '';
-			let embed;
-			const btn1 = [];
-			const btn2 = [];
-			const btn3 = [];
-			for (let n = 0 ; n < 15 ; n++) {
+			const EmbedDescriptionArray = [];
+			const selectMenuRow = new MessageActionRow();
+			const songSelectMenu = new MessageSelectMenu({
+				customID: `${this.name}`,
```
Comment (Ayeven) [inline slash/music/scSearch.js:32]: If you notice on the `interaction` event file where I write `const SlashCommand = SlashCommands.get(SlashCommandName) || SlashCommands.get(CustomID.split('_')[0]);` This line will need `${this.name}_selectMenusName` Well, classic string split hack.

File: slash/music/scSearch.js [hunk 2]
```diff
@@ -4,106 +4,86 @@ require('moment-duration-format');
...
-				btn3.push(
+				EmbedDescriptionArray.push(`${(n + 1).toString().padStart(2, '0')}) \`${duration}\` | **${result[n].title}**`);
+				songSelectMenu.addOptions([
 					{
-						type:'BUTTON',
-						label:`PLAY ${(n + 11).toString().padStart(2, '0')}`,
-						style:'DANGER',
-						customID:`${this.name}_${ n + 11}`,
+						label: `${(n + 1).toString().padStart(2, '0')}`,
+						description: `${duration} | ${result[n].title}]`.slice(0, 48),
```
Comment (Ayeven) [inline slash/music/scSearch.js:43]: What does `.slice(0, 48)` do here?

File: slash/music/scSearch.js [hunk 3]
```diff
@@ -4,106 +4,86 @@ require('moment-duration-format');
...
+     * @param {import('discord.js').SelectMenuInteraction} interaction - Represents a command interaction.
+     * @param {import('discord.js').Collection<bigint, Playlist>} playlist - List of song(s) in Discord.js Collection format
+     */
+	async selectmenu(interaction, playlist) {
 		try {
 			await interaction.deferUpdate();
 			let list = playlist.get(interaction.guildID);
 			const query = interaction.message.embeds[0].title;
-			const arrayResult = await scdl.search({ limit:30, resourceType:'tracks', query });
+			const arrayResult = await scdl.search({ limit: 30, resourceType: 'tracks', query });
```
Comment (Ayeven) [inline slash/music/scSearch.js:69]: I just realized, I put in limit 15 on `run` method while in button/select menu is 30. My bad on that

File: events/interaction.js [hunk 4]
```diff
@@ -1,23 +1,24 @@
...
-		if (!interaction.member) {return;}
+		if (!interaction.member) return;
+
 		const { SlashCommands } = client;
 		const SlashCommandName = interaction.commandName;
-		const ButtonID = interaction.customID;
-		const SlashCommand = (SlashCommands.get(SlashCommandName) || SlashCommands.get(ButtonID.split('_')[0]));
+		const CustomID = interaction.customID;
+		const SlashCommand = SlashCommands.get(SlashCommandName) || SlashCommands.get(CustomID.split('_')[0]);
 		const time = moment().format('YYYY MM DD HH:mm:ss');
```
Comment (Ayeven) [inline events/interaction.js:20]: Reviewing the code, I realize function `button` and `selectmenu` don't have them. so the `${time}` in both function is useless. Another oversight from me :kekw:

**Review by Ayeven — CHANGES_REQUESTED** [review]
Line 32 of `scSearch.js` need extra param as I commented

File: slash/music/scSearch.js (hunk 2 above)
Comment (Murky-MoMo) [reply slash/music/scSearch.js:43]: https://discord.com/developers/docs/interactions/message-components#select-menu-object-select-option-structure
select menu's description has a limit of 50 characters, so, to be safe, i cut the string down to 48 characters

File: slash/music/scSearch.js (hunk 1 above)
Comment (Murky-MoMo) [reply slash/music/scSearch.js:32]: select menu doesn't work like buttons
each buttons have different `customID`s
and here, there's different `value`s on a single select menu, so there's no need to add anything here
if you ever user multiple select menus, then you may need to add different IDs for them, but for now, this is completely functional with just that

File: slash/music/scSearch.js (hunk 3 above)
Comment (Murky-MoMo) [reply slash/music/scSearch.js:69]: while converting the code, i didn't notice that, yeah it doesn't really change the outcome, but it's more consistent to fetch 15
I'll also have to mention how select menus have a max of 25 options available

File: events/interaction.js (hunk 4 above)
Comment (Murky-MoMo) [reply events/interaction.js:20]: Sorry what?

File: slash/music/scSearch.js (hunk 1 above)
Comment (Ayeven) [reply slash/music/scSearch.js:32]: I tried to test the PR, but met with no select menu pop up. Asked around on `discord.js` discord server, everything seem right, I even double check documentation.  This is all I got. Not sure what really happened. 
image

Comment (Murky-MoMo) [reply slash/music/scSearch.js:32]: we should honestly add a debug feature like how discord.js uses it
however, i'll redownload and test here, maybe your discord apps outdated
for now, console log the message, if you see the select menu in <Message>.components, then it's your discord

Comment (Ayeven) [reply slash/music/scSearch.js:32]: F! All it need is a restart of discord app, and the select menu appeared. The most wasted 2hrs of my time trying to figure something simple

Comment (Murky-MoMo) [reply slash/music/scSearch.js:32]: aha.. so it was your discord app as I said
well is there anything else you need clarification on?

**Comment by Murky-MoMo** [comment]
> Line 32 of `scSearch.js` need extra param as I commented
yeah and I mentioned why it doesn't necessarily need it
the code works completely fine and since you're using a single component, which is the single select menu, it doesn't really need that
if you're using multiple components, then it'd be useful

**Pull request merged**